book/src/block-confirmation.md

```diff
@@ -0,0 +1,78 @@
+# Block Confirmation
+
+A validator votes on a PoH hash for two purposes. First, the vote indicates it
+believes the ledger is valid up until that point in time. Second, since many
+valid forks may exist at a given height, the vote also indicates exclusive
+support for the fork. This document describes only the former. The latter is
+described in [fork selection](fork-selection.md).
+
+## Current Design
+
+To start voting, a validator first registers an account to which it will send
+its votes. It then sends votes to that account. The vote contains the tick
+height of the block it is voting on. The account stores the 32 highest heights.
+
+### Problems
+
+* Only the validator knows how to find its own votes directly.
+
+  Other components, such as the one that calculates confirmation time, needs to
+  be baked into the fullnode code. The fullnode code queries the bank for all
+  accounts owned by the vote program.
+
+* Voting ballets do not contain a PoH hash. The validator is only voting that
+  it has observed an arbitrary block at some height.
+
+* Voting ballets do not contain a hash of the bank state. Without that hash,
+  there is no evidence that the validator executed the transactions and
+  verified there were no double spends.
+
+## Proposed Design
+
+### No Cross-block State Initially
+
+At the moment a block is produced, the leader shall add a NewBlock transaction
+to the ledger with a number of tokens that represents the validation reward.
+It is effectively an incremental multisig transaction that sends tokens from
+the mining pool to the validators. The account should allocate just enough
+space to collect the votes required to achieve a supermajority. When a
+validator observes the NewBlock transaction, it has the option to submit a vote
+that includes a hash of its ledger state (the bank state). Once the account has
+sufficient votes, the vote program should disperse the tokens to the
+validators, which causes the account to be deleted.
```

Comment:
to all validators, or only to those that voted on this NewBlock (i.e. the Ballot?)

Reply:
Good question. I don't know. Paging @ericlwilliams

Reply:
In the current design,  for each Solana 'epoch' (which we might want to discuss, I'm not sure it make sense to distribute on every block):
* All active validators are distributed mining pool rewards in proportional to their stake and % activity (e.g. uptime) during that epoch.   
* Leaders are able to keep the transaction fees collected during their rotations (within each epoch) _above_ a protocol-defined amount, which is returned to the mining pool.  This is alike a per-transaction fee tax that ensures long-term sustainability. 

This is outline in more detail in the working doc here (https://docs.google.com/document/d/1sntxE-o_WTgM-jrf8kkzKFEtbwx8sH84q4Bs28Ni5Zo/edit?usp=sharing),  soon to be moved to PR for the book. Feel free to dive into that doc (or wait for the PR) with any questions/comments/concerns.  Very high-level design right now, and we should be aligned on the major components.

Reply:
what about auto-delegating payments per block? i.e. rewards may be earned per block, but those rewards are 'locked up' with the rest of the stake, until a 'withdrawal' transaction is sent and after the appropriate cooling off time?

Reply:
@ericlwilliams, I'm not sure this answers Rob's question. "% activity", does that mean voting? And does the vote's state hash need to match the leader's state hash to get the reward? Incentive to agree seems bad, but to disagree takes no computation power (random hash), so that seems worse.

Reply:
To match the design, the payout should be to the validators that voted on that state, in proportion to their stake (out of the set that voted... I think). I think this implementation would result in "% activity" being the "% of states voted" on the surviving chain, which seems fine. 

Keep in mind, block payouts should, ideally, be a function of network status (% staked out of circulating and time since genesis block). This needs to be calculated at some 'epoch' level frequency and then those constants (e.g. block reward amount) to be payed out to the participating validators throughout that epoch.  

"And does the vote's state hash need to match the leader's state hash to get the reward?"  Does this just mean that a validator is voting on the state that the leader is proposing?  If so, then I don't see any other way to do it.  You should get rewarded if you bet on the correct state. I.e. you should get paid on the fork you vote on. I may not be following completely though, because I don't understand what you mean by disagreeing with a random hash (as opposed to just not voting).

Reply:
Interesting. I'm not familiar with the state proposal mechanism by validators.  My understanding was the leader proposes the latest state (for their TDMA slot), validators may  validate the proposed hash (i.e. compute and validate the state themselves), but then just have the option to cast a vote (or no-vote) on the fork with the hash that the leader is proposing (ignoring any delegated stake scenarios).  
So let's say, I'm a validator and I calculate a different hash (or random hash), how/where do I submit this?  Won't this just be ignored unless that validator is next leader? Is this submission channel separate from the voting mechanism (i.e. can I vote yes, but submit a different hash)?   I may be missing something obvious here, maybe just point me to the right section of the book :)

Reply:
So that part's just being implemented this week. Current plan was that validators submit their own hash and that the leader not reveal its hash until it's used for a supermajority.  The Yes/No vote that you proposed would work as well. We were thinking that hiding that hash would prevent validators from cheating (by always voting Yes and accepting the validation reward).  In hindsight, I think the argument for doing that is flawed, and that a simple Yes/No vote would be adequate as long as the slot leader randomly lies and then slashes any Yes votes.

Reply:
it occurs to me that as of today, leaders can't actually calculate the hash of the bank